### PR TITLE
fix(sql): fix spurious ambiguous column name error with sample by and `joins`

### PR DESCRIPTION
**Pull request by bluestreak01**
fixes https://github.com/questdb/questdb/issues/2625

### CODE REVIEW

File: core/src/main/java/io/questdb/griffin/SqlOptimiser.java [hunk 1]
```diff
@@ -3372,6 +3361,42 @@ private QueryModel rewriteSelectClause0(final QueryModel model, SqlExecutionCont
             }
         }
 
+        if (sampleBy != null && baseModel.getTimestamp() != null){
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/griffin/SqlOptimiser.java:3364]: nit:
```suggestion
        if (sampleBy != null && baseModel.getTimestamp() != null) {
```

P.S. Probably we need to configure a linter finally. My built-in linter may fail from time to time.

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 21 / 22 (95.45%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|20|21|95.24%|
|:large_blue_circle:|io/questdb/griffin/SqlException.java|1|1|100.00%|

File: core/src/main/java/io/questdb/griffin/SqlOptimiser.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/SqlOptimiser.java]: good spot!

I usually format code, but overlooked it this time 👍

Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/SqlOptimiser.java]: done in https://github.com/questdb/questdb/pull/2685/commits/1cc08eeb346ce406e7691d606ce3f06eb7436676

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 21 / 22 (95.45%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlOptimiser.java|20|21|95.24%|
|:large_blue_circle:|io/questdb/griffin/SqlException.java|1|1|100.00%|

**Pull request merged**